aips/aip-x.md

```diff
@@ -0,0 +1,105 @@
+  * (-) Some data structures cannot be deleted without potentially breaking applications on the blockchain, for example, a liquidity pool may be entirely broken if the fungible asset metadata were deleted. This could result in valuable assets being permanently locked on-chain without leveraging a governance proposal.
+
+The burn address solution was chosen as it provides a safe means to remove content associated with the users account without potentially breaking other applications.
+
+Note, the current proposal is for an `unburn` function to allow `primary_fungible_store` to reclaim accidentally transferred objects. This could instead be an alternative call into `primary_fungible_store` that explicitly has `unburn`, but that requires that wallets or other applications adopt the `unburn` functionality. The cost for checking the ownership should be negligible as the ownership must be checked prior to extracting assets.
+
+## Specification
+
+The burn address is defined as `0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff` or the largest valid unsigned 256-bit number. Upon acceptance of this AIP, a special Aptos name will be reserved for the burn address.
+
+The following new functions will be added to `aptos_framework::object`:
+
+`public entry fun burn<T: key>(owner: &signer, object: Object<T>)`
+
+This will verify that the `owner` is the actual owner of the object and then update the `ObjectCore::owner` field to the burn address.
+
+`public(friend) entry unburn<T: key>(original_owner: address, object: Object<T>)`
+
+This will verify that the current owner is the burn address and then update the `ObjectCore::owner` field to the `original_owner`.
+
+In `aptos_framework::primary_fungible_store`, the functions `transfers` and `withdraw` will check store ownership prior to transfers and call `object::unburn` if the owner is the burn address.
+
+## Reference Implementation
+
+## Testing 
+
+Thorough Move unit tests including:
+* Burning of objects, fungible assets, and digital assets.
+* Reclamation of primary fungible asset stores.
+* Reclamation of an asset by the owner of a `TransferRef`.
+* Ability to delete a burned asset with a `DeleteRef`.
+
+## Risks and Drawbacks
+
+There has been a lot of sentiment shared on the current state of receiving unwanted assets. Frankly speaking, no one wants a noob noob, except movekevin.
+
+This allows *soul bound* assets to effectively be decoupled. As a result, a user can potentially claim multiple *soul bound* assets. Of course, *soul bound* assets likely need to be monitored by an indexer that recognizes that a user may have already received an item even if it is no longer attached to their account.
+
+## Future Potential
+
+The future is a decentralized allow-list managed completely on-chain.
+
+## Timeline
+
+### Suggested implementation timeline
+
+This will be available before the 1.7 branch cut and available for inclusion in an earlier upgrade if the functionality is desired.
```

Comment:
Does this need to be urgently in 1.7?

Reply:
No, but I'm going to code it this afternoon, so it will be ready.